src/tagged-type.ts

```diff
@@ -261,9 +269,29 @@ const encode = (input: QueryValue): QueryValue => {
         return encodeMap["set"](input);
       } else if (input instanceof StreamToken) {
         return encodeMap["streamToken"](input);
+      } else if (input instanceof ArrayBuffer || ArrayBuffer.isView(input)) {
+        return encodeMap["bytes"](input);
       } else {
         return encodeMap["object"](input);
       }
   }
   // anything here would be unreachable code
 };
+
+function base64toBuffer(value: string): Uint8Array {
```

Comment:
Shouldn't this return an ArrayBuffer?

Reply:
I switched the decoding output to be Uint8Array based on feedback. We can rename these helper functions to "...Array..." instead of "...Buffer..." if that helps.

Reply:
If we decode to Uint8Array I do not think we should not accept any other TypedArray variant on encode. It's too easy to get send one thing and get a slightly different thing back. So accept only ArrayBuffers and Uint8Arrays.

Reply:
Done in [e06dabc](https://github.com/fauna/fauna-js/pull/260/commits/e06dabc0432a3d1cd6c0e17376897286ba6c558d)